packages/bigquery/src/table.js

```diff
@@ -1185,15 +1009,33 @@ Table.prototype.insert = function(rows, options, callback) {
     });
   }
 
-  delete options.raw;
+  delete json.raw;
+
+  var autoCreate = !!options.autoCreate;
+  var schema;
+
+  delete json.autoCreate;
+
+  if (autoCreate) {
+    if (!options.schema) {
+      throw new Error('Schema must be provided in order to auto-create Table.');
+    }
+
+    schema = options.schema;
+    delete json.schema;
+  }
 
   this.request({
     method: 'POST',
     uri: '/insertAll',
     json: json
   }, function(err, resp) {
     if (err) {
-      callback(err, resp);
+      if (err.code === 404 && autoCreate) {
+        setTimeout(createTableAndRetry, Math.random() * 60000);
```

Comment:
Why not create right away?

Reply:
https://github.com/GoogleCloudPlatform/google-cloud-ruby/issues/1635#issuecomment-327543698

Reply:
Thanks-- we should definitely link to that in a code comment as well.